engine/src/main/java/org/terasology/rendering/openvrprovider/OpenVRProvider.java

```diff
@@ -0,0 +1,407 @@
+    private static VR_IVRSystem_FnTable vrSystem;
+    private static VR_IVRCompositor_FnTable vrCompositor;
+    private static VR_IVROverlay_FnTable vrOverlay;
+    private static VR_IVRSettings_FnTable vrSettings;
+    private static int[] controllerDeviceIndex = new int[2];
+    private static VRControllerState_t.ByReference[] inputStateRefernceArray = new VRControllerState_t.ByReference[2];
+    private static VRControllerState_t[] controllerStateReference = new VRControllerState_t[2];
+    private static IntBuffer hmdErrorStore;
+    private static TrackedDevicePose_t.ByReference hmdTrackedDevicePoseReference;
+    private static TrackedDevicePose_t[] hmdTrackedDevicePoses;
+    private static boolean[] controllerTracking = new boolean[2];
+    //keyboard
+    private static boolean keyboardShowing;
+    private static boolean headIsTracking;
+
+
+    // TextureIDs of framebuffers for each eye
+    private final VRTextureBounds_t texBounds = new VRTextureBounds_t();
+    private float nearClip = 0.5f;
+    private float farClip = 500.0f;
+
+    public static Texture_t texType[] = new Texture_t[2];
+
+    public OpenVRState vrState = new OpenVRState();
+
+    public OpenVRProvider() {
+        for (int c = 0; c < 2; c++) {
+            controllerDeviceIndex[c] = -1;
+            controllerStateReference[c] = new VRControllerState_t();
+            inputStateRefernceArray[c] = new VRControllerState_t.ByReference();
+            inputStateRefernceArray[c].setAutoRead(false);
+            inputStateRefernceArray[c].setAutoWrite(false);
+            inputStateRefernceArray[c].setAutoSynch(false);
+            texType[c] = new Texture_t();
+        }
+    }
+
+    public boolean init() {
+        if (!initializeOpenVRLibrary()) {
+            return false;
+        }
+        if (!initializeJOpenVR())
+            return false;
+        int nAttempts = 0;
+        boolean bSuccess = false;
+
+        // OpenVR has a race condition here - it is necessary
+        // to initialize the overlay, but certain operations
+        // that appear to take place outside of the main thread
+        // seem to prevent that from happening if it's done too
+        // soon after OpenVR is initialized. This loop waits a
+        // reasonable amount of time and makes several attempts.
+        // In my testing, it works all of the time.
+        while (!bSuccess && nAttempts < 10) {
+            try {
+                Thread.sleep(300);
+            } catch(InterruptedException ex) {
+                Thread.currentThread().interrupt();
+            }
+            bSuccess = initOpenVRCompositor(true);
+            nAttempts++;
+        }
+        if (!initOpenVROverlay())
+            return false;
+        if (!initOpenVROSettings())
+            return false;
+        initialized = true;
+        return true;
+    }
+
+    public boolean isInitialized() {
+        return initialized;
+    }
+
+    public void shutdown() {
+        initialized = false;
```

Comment:
So, this has never actually caused a problem with program flow, but I guess I should explicitly plan for VRProvider to get deleted and re-initialized later. I'm going add VR_Shutdown() here and clean things up, and test that there are no problems with it.